Core/GDCore/IDE/PropertyFunctionGenerator.cpp

```diff
@@ -0,0 +1,245 @@
+        action.SetParametersCount(3);
+        action.SetParameter(0, "Object");
+        action.SetParameter(1, "Behavior");
+        action.SetParameter(2, "yes");
+        event.GetActions().Insert(action, 0);
+      }
+      {
+        auto &event =
+            dynamic_cast<gd::StandardEvent &>(setter.GetEvents().InsertNewEvent(
+                project, "BuiltinCommonInstructions::Standard", 0));
+
+        gd::Instruction condition;
+        condition.SetType("GetArgumentAsBoolean");
+        condition.SetParametersCount(1);
+        condition.SetParameter(0, "\"Value\"");
+        condition.SetInverted(true);
+        event.GetConditions().Insert(condition, 0);
+
+        gd::Instruction action;
+        action.SetType(setterType);
+        action.SetParametersCount(3);
+        action.SetParameter(0, "Object");
+        action.SetParameter(1, "Behavior");
+        action.SetParameter(2, "no");
+        event.GetActions().Insert(action, 0);
+      }
+    } else {
+      auto &event =
+          dynamic_cast<gd::StandardEvent &>(setter.GetEvents().InsertNewEvent(
+              project, "BuiltinCommonInstructions::Standard", 0));
+
+      gd::Instruction action;
+      action.SetType(setterType);
+      action.SetParametersCount(4);
+      action.SetParameter(0, "Object");
+      action.SetParameter(1, "Behavior");
+      action.SetParameter(2, "=");
+      action.SetParameter(3, "GetArgumentAs" + numberOrString + "(\"Value\")");
+      event.GetActions().Insert(action, 0);
+    }
+  }
+}
+
+bool PropertyFunctionGenerator::CanGenerateGetterAndSetter(
+    const gd::EventsBasedBehavior &eventsBasedBehavior,
+    const gd::NamedPropertyDescriptor &property) {
+  auto &type = property.GetType();
+  if (type != "Boolean" && type != "Number" && type != "String" &&
+      type != "Choice" && type != "Color") {
+    return false;
+  }
+
+  auto &functionsContainer = eventsBasedBehavior.GetEventsFunctions();
+  auto getterName = CapitalizeFirstLetter(property.GetName());
+  auto setterName = "Set" + getterName;
+  return !functionsContainer.HasEventsFunctionNamed(setterName) &&
+         !functionsContainer.HasEventsFunctionNamed(getterName);
+};
+
+gd::String PropertyFunctionGenerator::GetStringifiedExtraInfo(
+    const gd::PropertyDescriptor &property) {
+  if (property.GetType() == "Choice") {
+    gd::String arrayString;
+    arrayString += "[";
+    bool isFirst = true;
+    for (const gd::String &choice : property.GetExtraInfo()) {
+      if (!isFirst) {
+        arrayString += ",";
+      }
+      isFirst = false;
+      arrayString += "\"" + choice + "\"";
+    }
+    arrayString += "]";
+    return arrayString;
+  }
+  return "";
+}
+
+gd::String
+PropertyFunctionGenerator::CapitalizeFirstLetter(const gd::String &string) {
+  if (string.empty()) {
+    return string;
+  }
+  return string.substr(0, 1).UpperCase() + string.substr(1);
+}
+
+gd::String
+PropertyFunctionGenerator::UnCapitalizeFirstLetter(const gd::String &string) {
```

Comment:
Should those 2 functions be in a separate String helpers file? Or in the String class?